launch.go

```diff
@@ -182,11 +182,23 @@ func launch(api screwdriver.API, buildID string, rootDir string) error {
 		j.Name = "main"
 	}
 
-	err = gitSetup(scm.httpsString(), w.Src, pr, b.SHA)
+	repo, err := newRepo(scm.httpsString(), w.Src)
 	if err != nil {
 		return err
 	}
 
+	err = repo.Checkout(b.SHA)
+	if err != nil {
+		return err
+	}
+
+	if pr != "" {
+		err = repo.MergePR(pr, "pr")
```

Comment:
Sounds good! I'm thinking we still want to ensure merge takes a branch argument, so something like the following:  
```
func (r repo) MergePR(prNumber string) error {
    branch := "pr"
	if err := fetchPR(prNumber, branch, r.Path); err != nil {
		return fmt.Errorf("fetching pr: %v", err)
	}

	if err := merge(branch, r.Path); err != nil {
		return fmt.Errorf("merging pr: %v", err)
	}
	return nil
}

```